File: python/Film.py
# Class Film
class Film :
    # COnstructor (inisialisasi)
    def __init__(self, idFilm=0, judul="", genre="", durasi=""):
        self.__id = idFilm
        self.__judul = judul
        self.__genre = genre
        self.__durasi = durasi

    # Setter and Getter untuk id
    def setId(self, idFilm):
        self.__id = idFilm
    def getId(self):
        return self.__id

    # Setter and Getter untuk judul
    def setJudul(self, judul):
        self.__judul = judul
    def getJudul(self):
        return self.__judul

    # Setter and Getter untuk genre
    def setGenre(self, genre):
        self.__genre = genre
    def getGenre(self):
        return self.__genre

    # Setter and Getter untuk durasi
    def setDurasi(self, durasi):
        self.__durasi = durasi
    def getDurasi(self):
        return self.__durasi

File: php/Film.php
<?php
// Class Film untuk merepresentasikan objek film
class Film {
    // Atribut privat (encapsulation)
    private $id;
    private $judul;
    private $genre;
    private $durasi;
    private $gambar; // path file lokal, contoh: "uploads/spiderman.jpg"

    // Constructor
    public function __construct($id = 0, $judul = "", $genre = "", $durasi = "", $gambar = "") {
        $this->id     = $id;
        $this->judul  = $judul;
        $this->genre  = $genre;
        $this->durasi = $durasi;
        $this->gambar = $gambar;
    }

    // Getter & Setter ID
    public function setId($id){ $this->id = $id; }
    public function getId(){ return $this->id; }

    // Getter & Setter Judul
    public function setJudul($judul){ $this->judul = $judul; }
    public function getJudul(){ return $this->judul; }

    // Getter & Setter Genre
    public function setGenre($genre){ $this->genre = $genre; }
    public function getGenre(){ return $this->genre; }

    // Getter & Setter Durasi
    public function setDurasi($durasi){ $this->durasi = $durasi; }
    public function getDurasi(){ return $this->durasi; }

    // Getter & Setter Gambar
    public function setGambar($gambar){ $this->gambar = $gambar; }
    public function getGambar(){ return $this->gambar; }
}
?>